woocommerce/google-listings-and-ads
# Additional data points for tracker snapshot.

### Changes proposed in this Pull Request:

- Added additional data points for tracker snapshot.
- I modified the scope of has_at_least_one_synced_product() and has_account_issues method() to be able to pull data. 

Closes #1278.


Note: I had doubt about performance of [has_at_least_one_synced_product() ](https://github.com/woocommerce/google-listings-and-ads/blob/develop/src/Notes/AbstractSetupCampaign.php#:~:text=protected%20function%20has_at_least_one_synced_product()%3A%20bool%20%7B), but I can't immediately see an issue here. get_product_statistics() has a a type of caching mechanism.


### Detailed test instructions:

1. Add a code snippet to temporarily log the tracker data:
```
add_action(

	'admin_init',
	function() {
		$tracks = apply_filters( 'woocommerce_tracker_data', [] );
		error_log( json_encode( $tracks, JSON_PRETTY_PRINT ) );
	}
);
```

2. Refresh a page to confirm that the data is logged
3. Check the log file to confirm that the data is shown as expected
```
{
    "extensions": {
        "gla": {
            "settings": {
                "version": "1.11.1",
                "db_version": "1.11.1",
                "tos_accepted": "yes",
                "google_connected": "yes",
                "mc_setup": "yes",
                "ads_setup": "no",
                "target_audience": [
                    "ZA"
                ],
                "shipping_rate": "flat",
                "shipping_time": "flat",
                "offers_free_shipping": "no",
                "free_shipping_threshold": "",
                "tax_rate": "manual",
                "has_account_issue": "yes",
                "has_at_least_one_synced_product": "no",
                "ads_setup_started": "no"
            }
        }
    }
}
```

### Changelog entry

- Add  - Additional data points for tracker snapshot
